ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/PollableSubscriber.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import io.confluent.ksql.api.client.Row;
+import io.confluent.ksql.reactive.BaseSubscriber;
+import io.vertx.core.Context;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.reactivestreams.Subscription;
+
+public class PollableSubscriber extends BaseSubscriber<Row> {
+
+  private static final int REQUEST_BATCH_SIZE = 100;
+  // 100ms in ns
+  private static final long MAX_POLL_NANOS = TimeUnit.MILLISECONDS.toNanos(100);
+
+  private final BlockingQueue<Row> queue = new LinkedBlockingQueue<>();
+  private int tokens;
+  private Throwable error;
+  private volatile boolean closed;
+
+  public PollableSubscriber(final Context context) {
+    super(context);
+  }
+
+  @Override
+  protected void afterSubscribe(final Subscription subscription) {
+    checkRequestTokens();
+  }
+
+  @Override
+  protected void handleValue(final Row row) {
+    queue.add(row);
+  }
+
+  @Override
+  protected synchronized void handleError(final Throwable t) {
```

Comment:
Most probably shouldn't be synchronized otherwise it will block if polling is occurring at same time.

Reply:
Done.